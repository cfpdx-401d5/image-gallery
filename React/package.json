{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
