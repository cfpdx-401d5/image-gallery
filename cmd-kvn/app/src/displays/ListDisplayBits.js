import React, { PropTypes } from 'react';

import DeleteImageButton from '../DeleteImageButton';

function ListDisplayBits(props) {

  return ( 
    <li>
      <a href={props.url}>{props.title}</a>: {props.description} 
      <DeleteImageButton deleteImage={props.deleteImage} _id={props._id} />
    </li>
  );
}

ListDisplayBits.propTypes = {
  deleteImage: PropTypes.func.isRequired,
  // _id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default ListDisplayBits;

