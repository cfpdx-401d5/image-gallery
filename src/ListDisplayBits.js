import React, { PropTypes } from 'react';

function ListDisplayBits(props) {

  return <li key={props.id}><a href={props.url}>{props.title}</a>: {props.description}</li>;
}

ListDisplayBits.propTypes = {
  key: PropTypes.instanceOf(Date),
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default ListDisplayBits;

