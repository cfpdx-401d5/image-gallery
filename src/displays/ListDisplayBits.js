import React, { PropTypes } from 'react';

import DeleteImageButton from '../DeleteImageButton';

function ListDisplayBits(props) {

  return ( 
    <li>
      <a href={props.url}>{props.title}</a>: {props.description} 
      <DeleteImageButton deleteImage={props.deleteImage} id={props.id} />
    </li>
  );
}

ListDisplayBits.propTypes = {
  deleteImage: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default ListDisplayBits;

