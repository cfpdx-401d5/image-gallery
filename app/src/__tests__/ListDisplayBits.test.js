/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ListDisplayBits from '../displays/ListDisplayBits';

function testFn(a) {
    return a;
}

describe('ListDisplayBits component', function () {
    it('renders the ListDisplayBits component', () => {
        const rendered = shallow(<ListDisplayBits deleteImage={testFn} id='_someId' url='_url' title='_title' description='desc'/>);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});