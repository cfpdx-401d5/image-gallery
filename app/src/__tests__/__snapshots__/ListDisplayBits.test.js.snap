exports[`ListDisplayBits component renders the ListDisplayBits component 1`] = `
<li>
  <a
    href="_url">
    _title
  </a>
  : 
  desc
  <DeleteImageButton
    deleteImage={[Function]}
    id="_someId" />
</li>
`;
